src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,178 @@
+        } else {
+            System.out.println("Bye! Hope to see you again soon!");
+            return "Bye";
+        }
+    }
+
+    /**
+     * Displays the user's current task list.
+     *
+     * @param tasks The current TaskList of the User.
+     */
+    public String list(TaskList tasks, boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            if (tasks.size() == 0) {
+                String noTasks = "There are no tasks in your list yet.";
+                return noTasks;
+            } else {
+                String response = "Here are the tasks in your list:";
+                for (int i = 0; i < tasks.getTasks().size(); i++) {
+                    response += "\n" + Integer.toString((i + 1)) + ". " + tasks.getTasks().get(i).recordString();
+                }
+                return response;
+            }
+        } else {
+            if (tasks.size() == 0) {
+                System.out.println("There are no tasks in your list yet.");
+                return "";
+            } else {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < tasks.getTasks().size(); i++) {
+                    System.out.println((i + 1) + ". " + tasks.getTasks().get(i).recordString());
+                }
+                return "";
+            }
+        }
+    }
+
+    /**
+     * Notifies the user that the specified task has been marked done.
+     *
+     * @param t The specified task that was marked done.
+     */
+    public String done(Task t, boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            String response = "Nice! I've marked this task as done:";
+            response += "\n" + t.recordString();
+            return response;
+        } else {
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(t.recordString());
+            return "";
+        }
+    }
+
+    /**
+     * Notifies the user that the specified task has been deleted.
+     * Notifies the user how many tasks are left in the current TaskList.
+     *
+     * @param t The specified task that was deleted.
+     * @param size The number of tasks in the TaskList.
+     */
+    public String delete(Task t, int size, boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            String response = "Noted. I've removed this task:";
+            response += "\n" + t.recordString();
+            response += "\n" + "Now, you have " + Integer.toString(size) + " tasks in the list";
+            return response;
+        } else {
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(t.recordString());
+            System.out.println("Now, you have " + size + " tasks in the list");
+            return "";
+        }
+    }
+
+    /**
+     * Notifies the user that the specified task has been added.
+     * Notifies the user how many tasks are in the updated TaskList.
```

Comment:
I noticed this is a repeat of the earlier issue. Perhaps we can combine the comment into one line to make the comment more succinct?